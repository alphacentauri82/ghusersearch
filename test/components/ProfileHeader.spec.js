import React from 'react';
import renderer from 'react-test-renderer';
import {
  StyleSheetTestUtils,
} from 'aphrodite';
import ProfileHeader from 'components/ProfileHeader';

describe('Component: ProfileHeader', () => {

  beforeAll(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterAll(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('should render', () => {
    const props = {
      avatar_url: 'url',
      bio: 'bio',
      blog: 'blog',
      company: 'company',
      followers: 20,
      following: 40,
      html_url: 'html_url',
      location: 'location',
      login: 'login',
      name: 'name',
      id: 123,
      public_repos: 12,
    };
    const component = renderer.create(
      <ProfileHeader {...props} />
    );
    expect(component).toMatchSnapshot();
  });

});
