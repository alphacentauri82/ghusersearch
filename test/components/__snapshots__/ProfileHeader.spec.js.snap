// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ProfileHeader should render 1`] = `
<div
  className="u-sm-flex"
>
  <div
    className="ProfileHeader_avatar_1ijjnm5"
  >
    <div
      className="Image_e296pg"
      style={
        Object {
          "minHeight": 60,
        }
      }
    >
      <img
        alt="name's avatar"
        className="Avatar_1pcpsid u-imgResponsive"
        onLoad={[Function]}
        src="url&s=460"
        width={230}
      />
      <svg />
    </div>
  </div>
  <div
    className="u-sm-flexGrow1 u-sm-flex u-sm-flexCol"
  >
    <div
      className="ProfileHeader_wrapTitle_1tdla59"
    >
      <div
        className="u-flex u-flexWrap u-flexAlignItemsBaseline"
      >
        <h1
          className="ProfileTitle_name_1oubu48"
        >
          name
        </h1>
        <p
          className="ProfileTitle_username_irnxte"
        >
          <a
            href="html_url"
          >
            @
            login
          </a>
        </p>
      </div>
    </div>
    <div
      className="ProfileHeader_wrapBio_1fopc40"
    >
      <p
        className="Bio_qh35s3"
        dangerouslySetInnerHTML={
          Object {
            "__html": "bio",
          }
        }
      />
    </div>
    <div
      className="ProfileHeader_wrapUserInfo_koa90h"
    >
      <ul>
        <li
          className="UserInfo_item_1bldfdq u-flex u-flexAlignItemsCenter"
        >
          <svg />
          <p
            className="UserInfo_text_f9e8qy"
            dangerouslySetInnerHTML={
              Object {
                "__html": "company",
              }
            }
          />
        </li>
        <li
          className="UserInfo_item_1bldfdq u-flex u-flexAlignItemsCenter"
        >
          <svg />
          <p
            className="UserInfo_text_f9e8qy"
          >
            location
          </p>
        </li>
        <li
          className="UserInfo_item_1bldfdq u-flex u-flexAlignItemsCenter"
        >
          <svg />
          <p
            className="UserInfo_text_f9e8qy"
          >
            <a
              href="blog"
            >
              blog
            </a>
          </p>
        </li>
      </ul>
    </div>
    <div
      className="u-sm-flexExpandTop"
    >
      <ul
        className="u-flex"
      >
        <li
          className="Stats_item_1p0a99 u-flex u-flexCol u-flexAlignItemsCenter"
        >
          <p
            className="Stats_value_176gzik"
          >
            20
          </p>
          <p
            className="Stats_label_f9e8qy"
          >
            Followers
          </p>
        </li>
        <li
          className="Stats_item_1p0a99 u-flex u-flexCol u-flexAlignItemsCenter"
        >
          <p
            className="Stats_value_176gzik"
          >
            40
          </p>
          <p
            className="Stats_label_f9e8qy"
          >
            Following
          </p>
        </li>
        <li
          className="Stats_item_1p0a99 u-flex u-flexCol u-flexAlignItemsCenter"
        >
          <p
            className="Stats_value_176gzik"
          >
            12
          </p>
          <p
            className="Stats_label_f9e8qy"
          >
            Repos
          </p>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
